mergify_engine/tests/functional/actions/test_rebase.py

```diff
@@ -0,0 +1,59 @@
+# -*- encoding: utf-8 -*-
+#
+#  Copyright © 2021 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import yaml
+
+from mergify_engine.tests.functional import base
+
+
+class TestRebaseAction(base.FunctionalTestBase):
+    async def test_rebase_ok(self):
+        rules = {
+            "pull_request_rules": [
+                {
+                    "name": "rebase",
+                    "conditions": [f"base={self.master_branch_name}"],
+                    "actions": {"rebase": {}},
+                }
+            ]
+        }
+
+        ahead_branch = self.get_full_branch_name("main/head")
+
+        await self.setup_repo(
+            yaml.dump(rules),
+            test_branches=[
+                ahead_branch,
+            ],
+        )
+
+        await self.wait_for("push", {"ref": f"refs/heads/{ahead_branch}"})
+
+        p, commits = await self.create_pr()
+        pr_initial_sha = commits[-1].sha
+
+        await self.git("checkout", ahead_branch)
+
+        open(self.git.tmp + "/an_other_file", "wb").close()
+        await self.git("add", "an_other_file")
+
+        await self.git("commit", "--no-edit", "-m", "commit ahead")
+        await self.git("push", "--quiet", "main", self.master_branch_name)
```

Comment:
The `wait_for("push")` should be here.

Reply:
That's my problem when I put the `wait_for()` line 49 or line 51, I raise this exception:

```
raise Exception(
            f"Never got event `{event_type}` with payload `{expected_payload}` (timeout)"
        )
E       Exception: Never got event `push` with payload `{'ref': 'refs/heads/20210120134511/test_rebase_ok/main/head'}` (timeout)

mergify_engine/tests/functional/base.py:183: Exception
```

Reply:
Why are you trying L49 or L51 while @sileht suggested L53? 🤔 
Did you try L53?

Reply:
same

Reply:
I have just refreshed my code if you want more details 👀